modules/bridge/src/main/ts/ephox/bridge/components/view/ViewButton.ts

```diff
@@ -1,37 +1,107 @@
 import { FieldSchema, StructureSchema } from '@ephox/boulder';
-import { Result } from '@ephox/katamari';
+import { Optional, Result } from '@ephox/katamari';
 
 import * as ComponentSchema from '../../core/ComponentSchema';
 
-export interface ViewNormalButtonSpec {
-  type: 'button';
+interface BaseButtonSpec {
   text: string;
+  tooltip?: string;
   buttonType?: 'primary' | 'secondary';
-  onAction: () => void;
+  borderless?: boolean;
 }
 
-export type ViewButtonSpec = ViewNormalButtonSpec;
+export interface TogglableButtonApi {
+  isActive: () => boolean;
+  setActive: (state: boolean) => void;
+  setIcon: (newIcon: string) => void;
+}
 
-export interface ViewNormalButton {
+export interface ViewNormalButtonSpec extends BaseButtonSpec {
   type: 'button';
+  onAction: () => void;
+}
+
+export interface ViewTogglableIconButtonSpec extends Omit<BaseButtonSpec, 'text'> {
+  text?: string;
+  name: string;
+  type: 'togglableIconButton';
+  tooltip: string;
+  icon: string;
+  onAction: (api: TogglableButtonApi) => void;
+}
+
+export interface ViewButtonsGroupSpec {
+  type: 'group';
+  buttons: Array<ViewNormalButtonSpec | ViewTogglableIconButtonSpec>;
+}
+
+export type ViewButtonSpec = ViewNormalButtonSpec | ViewTogglableIconButtonSpec | ViewButtonsGroupSpec;
+
+interface BaseButton {
   text: string;
+  tooltip: Optional<string>;
   buttonType: 'primary' | 'secondary';
+  borderless: boolean;
+}
+
+export interface ViewNormalButton extends BaseButton {
+  type: 'button';
   onAction: () => void;
 }
 
-export type ViewButton = ViewNormalButton;
+export interface ViewTogglableIconButton extends Omit<BaseButton, 'text' | 'tooltip'> {
```

Comment:
Why not `ViewTogglableIconButton` => `ViewTogglableButton`? since text, icon, and both are possible

Reply:
actually with the last implementation `icon` is mandatory, and it is the only part that at the moment is possible to toggle since this button does not expose any function to change the text for example